Error from darwin:
```
error: builder for '/nix/store/8xh0ayff2s5qrgvifjcgbxbccddlqp42-skia-124-unstable-2024-05-22.drv' failed with exit code 1;
       last 10 log lines:
       > FAILED: obj/src/gpu/gpu_shared.MutableTextureState.o
       > c++ -MD -MF obj/src/gpu/gpu_shared.MutableTextureState.o.d -DNDEBUG -DSK_CODEC_DECODES_BMP -DSK_CODEC_DECODES_WBMP -DSKIA_DLL -DSK_ENABLE_PRECOMPILE -DSK_ASSUME_GL=1 -DSK_GANESH -DSK_DISABLE_TRACING -DSK_USE_PERFETTO -DSK_ENABLE_API_AVAILABLE -DSK_GAMMA_APPLY_TO_A8 -DSK_ENABLE_AVX512_OPTS -DSKIA_IMPLEMENTATION=1 -I.. -Wno-attributes -ffp-contract=off -fstrict-aliasing -fPIC -fvisibility=hidden -march=armv7-a -mfpu=neon -mthumb -isysroot /nix/store/12rrl7arpahm7l708pvg1ly247sc0djg-SDKs/MacOSX11.0.sdk -target x86_64-apple-macos10.13 -O3 -I/nix/store/lha2hfsmgz1a5yrscpqaq9yfphjvi1g7-harfbuzz-icu-8.4.0-dev/include/harfbuzz -std=c++17 -fvisibility-inlines-hidden -fno-exceptions -fno-rtti -c ../src/gpu/MutableTextureState.cpp -o obj/src/gpu/gpu_shared.MutableTextureState.o
       > clang-16: warning: overriding '-mmacos-version-min=11.0' option with '-target x86_64-apple-macos10.13' [-Woverriding-t-option]
       > clang-16: error: unsupported argument 'armv7-a' to option '-march='
       > [8/1004] compile ../src/gpu/PipelineUtils.cpp
       > FAILED: obj/src/gpu/gpu_shared.PipelineUtils.o
       > c++ -MD -MF obj/src/gpu/gpu_shared.PipelineUtils.o.d -DNDEBUG -DSK_CODEC_DECODES_BMP -DSK_CODEC_DECODES_WBMP -DSKIA_DLL -DSK_ENABLE_PRECOMPILE -DSK_ASSUME_GL=1 -DSK_GANESH -DSK_DISABLE_TRACING -DSK_USE_PERFETTO -DSK_ENABLE_API_AVAILABLE -DSK_GAMMA_APPLY_TO_A8 -DSK_ENABLE_AVX512_OPTS -DSKIA_IMPLEMENTATION=1 -I.. -Wno-attributes -ffp-contract=off -fstrict-aliasing -fPIC -fvisibility=hidden -march=armv7-a -mfpu=neon -mthumb -isysroot /nix/store/12rrl7arpahm7l708pvg1ly247sc0djg-SDKs/MacOSX11.0.sdk -target x86_64-apple-macos10.13 -O3 -I/nix/store/lha2hfsmgz1a5yrscpqaq9yfphjvi1g7-harfbuzz-icu-8.4.0-dev/include/harfbuzz -std=c++17 -fvisibility-inlines-hidden -fno-exceptions -fno-rtti -c ../src/gpu/PipelineUtils.cpp -o obj/src/gpu/gpu_shared.PipelineUtils.o
       > clang-16: warning: overriding '-mmacos-version-min=11.0' option with '-target x86_64-apple-macos10.13' [-Woverriding-t-option]
       > clang-16: error: unsupported argument 'armv7-a' to option '-march='
       > ninja: build stopped: subcommand failed.
       For full logs, run 'nix-store -l /nix/store/8xh0ayff2s5qrgvifjcgbxbccddlqp42-skia-124-unstable-2024-05-22.drv'.
error: 1 dependencies of derivation '/nix/store/xvvwhcl9572fhc5q8w4dzxn61lhg2yz1-review-shell.drv' failed to build
```

`armv7-a` seems wrong? Maybe darwin support should come post-merge?

--- a/pkgs/by-name/sk/skia/package.nix
+++ b/pkgs/by-name/sk/skia/package.nix
@@ -1,0 +1,143 @@
+{ lib
+, stdenv
+, fetchgit
+, expat
+, fontconfig
+, freetype
+, harfbuzzFull
+, icu
+, gn
+, libGL
+, libjpeg
+, libwebp
+, libX11
+, ninja
+, python3
+, testers
+, vulkan-headers
+, vulkan-memory-allocator
+, xcbuild
+
+, enableVulkan ? !stdenv.isDarwin
+}:
+
+stdenv.mkDerivation (finalAttrs: {
+  pname = "skia";
+  # Version from https://skia.googlesource.com/skia/+/refs/heads/main/RELEASE_NOTES.md
+  # or https://chromiumdash.appspot.com/releases
+  # plus date of the tip of the corresponding chrome/m$version branch
+  version = "124-unstable-2024-05-22";
+
+  src = fetchgit {
+    url = "https://skia.googlesource.com/skia.git";
+    # Tip of the chrome/m$version branch
+    rev = "a747f7ea37db6ea3871816dbaf2eb41b5776c826";
+    hash = "sha256-zHfv4OZK/nVJc2rl+dBSCc4f6qndpAKcFZtThw06+LY=";
+  };
+
+  postPatch = ''
+    # System zlib detection bug workaround
+    substituteInPlace BUILD.gn \
+      --replace-fail 'deps = [ "//third_party/zlib" ]' 'deps = []'
+  '';
+
+  nativeBuildInputs = [
+    gn
+    ninja
+    python3
+  ] ++ lib.optional stdenv.isDarwin xcbuild;
+
+  buildInputs = [
+    expat
+    fontconfig
+    freetype
+    harfbuzzFull
+    icu
+    libGL
+    libjpeg
+    libwebp
+    libX11
+  ] ++ lib.optionals enableVulkan [
+    vulkan-headers
+    vulkan-memory-allocator
+  ];
+
+  configurePhase = ''
+    runHook preConfigure
+    gn gen build --args='${toString ([
+      # Build in release mode
+      "is_official_build=true"
+      "is_component_build=true"
+      # Don't use missing tools
+      "skia_use_dng_sdk=false"
+      "skia_use_wuffs=false"
+      # Use system dependencies
+      "extra_cflags=[\"-I${harfbuzzFull.dev}/include/harfbuzz\"]"
+    ] ++ map (lib: "skia_use_system_${lib}=true") [
+      "zlib"
+      "harfbuzz"
+      "libpng"
+      "libwebp"
+    ] ++ lib.optionals enableVulkan [
+      "skia_use_vulkan=true"
+    ])}'
+    cd build
+    runHook postConfigure
+  '';
+
+  # Somewhat arbitrary, but similar to what other distros are doing
+  installPhase = ''
+    runHook preInstall
+
+    # Libraries
+    mkdir -p $out/lib
+    cp *.so *.a $out/lib
+
+    # Includes
+    pushd ../include
+    find . -name '*.h' -exec install -Dm644 {} $out/include/skia/{} \;
+    popd
+    pushd ../modules
+    find . -name '*.h' -exec install -Dm644 {} $out/include/skia/modules/{} \;
+    popd
+
+    # Pkg-config
+    mkdir -p $out/lib/pkgconfig
+    cat > $out/lib/pkgconfig/skia.pc <<'EOF'
+    prefix=${placeholder "out"}
+    exec_prefix=''${prefix}
+    libdir=''${prefix}/lib
+    includedir=''${prefix}/include/skia
+    Name: skia
+    Description: 2D graphic library for drawing text, geometries and images.
+    URL: https://skia.org/
+    Version: ${lib.versions.major finalAttrs.version}
+    Libs: -L''${libdir} -lskia
+    Cflags: -I''${includedir}
+    EOF
+
+    runHook postInstall
+  '';
+
+  preFixup = ''
+    # Some skia includes are assumed to be under an include sub directory by
+    # other includes
+    for file in $(grep -rl '#include "include/' $out/include); do
+      substituteInPlace "$file" \
+        --replace-fail '#include "include/' '#include "'
+    done
+  '';
+
+  passthru.tests.pkg-config = testers.testMetaPkgConfig finalAttrs.finalPackage;
+
+  meta = {
+    description = "2D graphic library for drawing text, geometries and images";
+    homepage = "https://skia.org/";
+    license = lib.licenses.bsd3;
+    maintainers = with lib.maintainers; [ fgaz ];
+    platforms = lib.platforms.all;
+    pkgConfigModules = [ "skia" ];
+    # https://github.com/NixOS/nixpkgs/pull/325871#issuecomment-2220610016
+    broken = stdenv.isDarwin;
+  };
+})

